### PR TITLE
Added support for comment for operations

### PR DESCRIPTION
**Pull request by rozza**
JAVA-4369

### CODE REVIEW

File: driver-core/src/main/com/mongodb/client/model/InsertManyOptions.java
Comment (rozza) [inline driver-core/src/main/com/mongodb/client/model/InsertManyOptions.java:71]: Took the opportunity to document this is ignored in bulk operations.

File: driver-core/src/main/com/mongodb/client/model/BulkWriteOptions.java
Comment (rozza) [inline driver-core/src/main/com/mongodb/client/model/BulkWriteOptions.java:102]: Opted to add both `comment(String)` and `comment(BsonValue)` throughout - to keep the api's the consistent. Also, I took the opinion that most users won't be familiar with BsonValues.

File: driver-core/src/main/com/mongodb/internal/operation/AggregateOperation.java
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/operation/AggregateOperation.java:211]: This seems like it was written for public consumption, but this is an internal class.

Can you clarify your intent?

File: driver-core/src/main/com/mongodb/internal/operation/FindAndUpdateOperation.java
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/operation/FindAndUpdateOperation.java:74]: Could this go in BaseFindAndModifyOperation?

File: driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java:483]: Why `null` instead of `comment`, like it is for async just below?

File: driver-core/src/main/com/mongodb/internal/operation/FindAndUpdateOperation.java
Comment (rozza) [reply driver-core/src/main/com/mongodb/internal/operation/FindAndUpdateOperation.java:74]: I followed the existing conventions - like with hint

File: driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java
Comment (rozza) [reply driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java:483]: Good catch :+1:

File: driver-core/src/main/com/mongodb/internal/operation/AggregateOperation.java
Comment (rozza) [reply driver-core/src/main/com/mongodb/internal/operation/AggregateOperation.java:211]: Happy to remove if required.

File: driver-core/src/main/com/mongodb/internal/operation/FindAndUpdateOperation.java
Comment (rozza) [reply driver-core/src/main/com/mongodb/internal/operation/FindAndUpdateOperation.java:74]: Added JAVA-4544 which I'll do after this ticket :+1:

File: driver-core/src/main/com/mongodb/internal/operation/AggregateOperation.java
Comment (rozza) [reply driver-core/src/main/com/mongodb/internal/operation/AggregateOperation.java:211]: I had it to document the behavior of strings vs other types. Will remove.

**Comment by rozza** [comment]
Note some tests are failing - I am speaking with the authors as it seems the expected pre 4.4 failures don't fail on 3.2, 3.0 and 2.6

File: driver-core/src/main/com/mongodb/internal/operation/FindAndUpdateOperation.java
Comment (jyemin) [reply driver-core/src/main/com/mongodb/internal/operation/FindAndUpdateOperation.java:74]: No worries.

**Review by jyemin — COMMENTED** [review]
I'm unable to comment on some of the threads.  I don't know why.

I'll wait to LGTM until we figure out why tests are failing.

File: driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java
Comment (jyemin) [reply driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java]: Can I deduce that there is no test coverage for this scenario?

Comment (rozza) [reply driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java]: No - it would be for a getMore on a list collection / list indexes operation.

**Review by jyemin — APPROVED** [review]
LGTM if tests are all passing.

I don't think this needs a second review.

**Pull request closed without merging**
